ksql-execution/src/main/java/io/confluent/ksql/execution/function/udtf/KudtfFlatMapper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.function.udtf;
+
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.expression.tree.ColumnReferenceExp;
+import io.confluent.ksql.execution.expression.tree.FunctionCall;
+import io.confluent.ksql.execution.function.UdtfUtil;
+import io.confluent.ksql.execution.plan.StreamFlatMap;
+import io.confluent.ksql.function.KsqlTableFunction;
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.schema.ksql.ColumnRef;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.OptionalInt;
+import org.apache.kafka.streams.kstream.ValueMapper;
+
+public class KudtfFlatMapper implements ValueMapper<GenericRow, Iterable<GenericRow>> {
+
+  private final StreamFlatMap streamFlatMap;
+
+  public KudtfFlatMapper(final StreamFlatMap streamFlatMap) {
+    this.streamFlatMap = streamFlatMap;
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public Iterable<GenericRow> apply(final GenericRow row) {
+
+    /*
+    TODO this currently assumes function expression is a simple column ref expression
```

Comment:
Can we keep TODOs out of the commited code please?  Either complete the work, or keep them tracked locally, or add github issues to track the outstanding work.

Reply:
We should enable checkstyle rule for this

Reply:
While I agree that TODOs shouldn't remain in the code once the task (i.e. the entire feature/epic) has been completed, until it gets to that point I'd prefer to see TODOs than not, as they make it explicit that this is a WIP. This makes it more obvious about what is going on to someone who stumbles across the code and doesn't know the bigger context.

Reply:
I'm happy with TODOs in a feature branch.  But strongly against them in the main code base.

Reply:
removed